// range-input-react/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Range = function (_React$Component) {
  (0, _inherits3.default)(Range, _React$Component);

  function Range() {
    (0, _classCallCheck3.default)(this, Range);
    return (0, _possibleConstructorReturn3.default)(this, (0, _getPrototypeOf2.default)(Range).apply(this, arguments));
  }

  (0, _createClass3.default)(Range, [{
    key: 'onRangeChange',
    value: function onRangeChange(e) {
      e.persist();
      this.props.onMouseMove(e);
      if (e.buttons !== 1 && e.which !== 1) return;
      this.props.onChange(e);
    }
  }, {
    key: 'onRangeClick',
    value: function onRangeClick(e) {
      this.props.onClick(e);
      this.props.onChange(e);
    }
  }, {
    key: 'onRangeKeyDown',
    value: function onRangeKeyDown(e) {
      this.props.onKeyDown(e);
      this.props.onChange(e);
    }
  }, {
    key: 'render',
    value: function render() {
      var inputClass = this.props.className.concat(' input-range');
      return _react2.default.createElement('input', {
        className: inputClass,
        type: 'range',
        onChange: this.props.onChange.bind(this),
        onClick: this.props.onClick.bind(this),
        onKeyDown: this.props.onKeyDown.bind(this),
        onMouseMove: this.props.onMouseMove.bind(this),
        max: this.props.max,
        min: this.props.min,
        value: this.props.value
      });
    }
  }]);
  return Range;
}(_react2.default.Component);

Range.defaultProps = {
  className: 'range-default',
  onChange: function onChange() {},
  // ES6 method shorthand syntax
  onClick: function onClick() {},
  onKeyDown: function onKeyDown() {},
  onMouseMove: function onMouseMove() {}
};

Range.propTypes = {
  max: _react2.default.PropTypes.number,
  min: _react2.default.PropTypes.number,
  value: _react2.default.PropTypes.number,
  className: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func,
  onClick: _react2.default.PropTypes.func,
  onKeyDown: _react2.default.PropTypes.func,
  onMouseMove: _react2.default.PropTypes.func
};

exports.default = Range;
